test-utils/test-framework/src/yagna.rs

```diff
@@ -0,0 +1,232 @@
+            command.env(key, val);
+        }
+        Ok(command)
+    }
+
+    pub fn build_std(&self, binary: &str) -> std::process::Command {
+        let mut command = std::process::Command::new(cargo_bin(binary));
+        for (key, val) in &self.env {
+            command.env(key, val);
+        }
+        command
+    }
+
+    pub async fn service_run(self) {}
+}
+
+impl YagnaMock {
+    pub fn new(test_dir: &Path) -> Self {
+        let yagna = YagnaMock {
+            yagna_dir: test_dir.join("yagna"),
+            provider_dir: test_dir.join("provider"),
+            command: YagnaCommand {
+                env: Default::default(),
+            },
+            process: Arc::new(Mutex::new(None)),
+        };
+
+        yagna.set_default_env()
+    }
+
+    fn set_default_env(mut self) -> Self {
+        let gsb_url = format!("unix://{}", self.yagna_dir.join("gsb.sock").display());
+
+        self.command = self
+            .command
+            .env("YAGNA_DATADIR", self.yagna_dir.to_string_lossy())
+            .env("DATADIR", self.provider_dir.to_string_lossy())
+            .env("YAGNA_API_URL", "http://127.0.0.1:10000")
+            .env("YA_NET_TYPE", "hybrid")
+            .env("YA_NET_BIND_URL", "udp://0.0.0.0:0")
+            .env("YA_NET_RELAY_HOST", "yacn2a.dev.golem.network:7477")
+            .env("GSB_URL", gsb_url);
+        self
+    }
+
+    pub fn command(&self) -> Command {
+        self.command.build("yagna").unwrap()
+    }
+
+    pub async fn service_run(self) -> anyhow::Result<Self> {
+        let mut cmd = self.command.build_std("yagna");
+        cmd.current_dir(&self.yagna_dir)
+            .arg("service")
+            .arg("run")
+            .stderr(Stdio::null())
+            .stdout(Stdio::null());
+
+        #[cfg(target_os = "linux")]
+        unsafe {
+            use ::nix::libc::{prctl, PR_SET_PDEATHSIG};
+            use ::nix::sys::signal::*;
+            use std::io;
+            use std::os::unix::process::CommandExt;
+
+            cmd.pre_exec(|| {
+                ::nix::unistd::setsid().map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+                let _ = prctl(PR_SET_PDEATHSIG, SIGTERM);
+                Ok(())
+            });
+        }
+
+        let mut tracker = tracker::YagnaTracker::new(&mut cmd, &self.yagna_dir)?;
+        tracker.start().await?;
+
+        *self.process.lock().unwrap() = Some(tracker);
+        Ok(self)
+    }
+
+    pub(crate) async fn tear_down(&self, timeout: std::time::Duration) -> anyhow::Result<()> {
+        let process = {
+            self.process
+                .lock()
+                .map_err(|e| anyhow!("{e}"))?
+                .as_ref()
+                .map(|tracker| tracker.child.clone())
+        };
+        if let Some(process) = process {
+            if process.terminate(timeout).await.is_err() {
+                process.kill();
+            }
+        }
+        Ok(())
+    }
+}
+
+#[cfg(unix)]
+mod tracker {
```

Comment:
Consider using existing method for this: https://docs.rs/tokio/latest/tokio/process/struct.Command.html#method.kill_on_drop